File: Algorithms/Quicksort/Quicksort.cpp
template <class T>
void quicksort(T arr[], int low, int high)
{
    if (low >= high)
        return;

    int left = low, right = high, pivot = arr[(high + low) / 2];
    while (left <= right)
    {
        while (arr[left] < pivot)
            left++;
        while (arr[right] > pivot)
            right--;

        int temp = arr[left];
        arr[left++] = arr[right];
        arr[right--] = temp;
    }

    quicksort<T>(arr, low, right);
    quicksort<T>(arr, left, high);
}

File: Algorithms/Quicksort/Quicksort.py
def quicksort(arr, n):
    if n <= 1:
        return

    left, right, pivot = 0, n - 1, arr[n // 2]
    while left <= right:
        while arr[left] < pivot:
            left += 1
        while arr[right] > pivot:
            right -= 1

        arr[left], arr[right] = arr[right], arr[left]
        left += 1
        right -= 1

    quicksort(arr[: right + 1], right + 1)
    quicksort(arr[left:], n - left)
